### PR TITLE
Updates lidgren, adds ipv6 support

### PR DESCRIPTION
**Pull request by gbasood**
Implements IPv6 support along with the fixes to it from https://github.com/lidgren/lidgren-network-gen3/pull/33 and its discussion.

Also we were a few *years* behind on our version of lidgren.

### CODE REVIEW

**Comment by gbasood** [comment]
~Currently has two errors I haven't looked into resolving:~
```
'SS14NetServer' does not implement interface member 'ISS14NetServer.RegisterReceivedCallback(SendOrPostCallback)'
'SS14NetServer' does not implement interface member 'ISS14NetServer.SendMessage(NetOutgoingMessage, List<NetConnection>, NetDeliveryMethod, int)'	
```
This also made a lot of warnings show up on build, not sure what changed in the solution/csproj files to cause that.

**Comment by gbasood** [comment]
New problem:

Sending a chat message caused a crash.

**Comment by gbasood** [comment]
fixed above issue
something weird happened between the first compile and this one
long story short, the methods i thought I needed to add to satisfy interface requirements weren't actually doing that and instead were just becoming self-calling infinite recursion-causing methods.

Now we just need two people to test if ipv6 works

**Comment by gbasood** [comment]
server gets Connect message properly, tries to send connectresponse
client doesn't seem to get it or handle it, either or

**Comment by psykzz** [comment]
Just double checking this to the PR you listed, it looks like your doing things differently
Can you explain the changes?

For example in the PR he converts all IPs to ipv6 regardless. Is that not what your doing here?

**Comment by gbasood** [comment]
it's possible that my entire branch is fucked because it doesnt seem like the files match the ones in the original ipv6 fix branch

**Comment by gbasood** [comment]
This seems to be functional now
I need someone to test it for me on their own
Couple of notes for review:
 - git diff is significantly smaller with whitespace ignored
 - I had to redo the entire branch because I messed up the history at one point and was working on some weird hybrid of old and new lidgren
 - If we continue working with lidgren beyond this we might want to consider setting this up as a submodule

File: SS14.Server/Network/SS14NetServer.cs [hunk 1]
```diff
@@ -28,7 +30,8 @@ public void SendMessage(NetOutgoingMessage message, NetConnection client)
 
         public void SendToMany(NetOutgoingMessage message, List<NetConnection> recipients)
         {
-            SendMessage(message, recipients, NetDeliveryMethod.ReliableOrdered, 0);
+            foreach (NetConnection recipient in recipients)
```
Comment (gbasood) [inline SS14.Server/Network/SS14NetServer.cs]: Someone should tell me if this is hacky

File: SS14.Server/Network/SS14NetServer.cs [hunk 2]
```diff
@@ -43,5 +46,18 @@ public static NetPeerConfiguration LoadNetPeerConfig()
 
             return _config;
         }
+
+        public void SendMessage(NetOutgoingMessage message, List<NetConnection> recipients, NetDeliveryMethod method, int sequenceChannel)
+        {
+            foreach (NetConnection recipient in recipients)
```
Comment (gbasood) [inline SS14.Server/Network/SS14NetServer.cs:51]: Someone should tell me if this is hacky

File: SS14.Server/Network/SS14NetServer.cs (hunk 1 above)
Comment (PJB3005) [reply SS14.Server/Network/SS14NetServer.cs]: Shouldn't this just call the overload you commented on next?

**Comment by PJB3005** [comment]
@gbasood status on this?

**Comment by gbasood** [comment]
Right now clients get an exception on trying to access a closed filestream when they send a chat message, and I'd assume when they receive one too
Haven't looked into it much, I thought it wasn't part of my changes until this week

File: SpaceStation14.sln [hunk 3]
```diff
@@ -24,18 +24,16 @@ Global
...
+		{59250BAF-0000-0000-0000-000000000000}.Debug|Any CPU.ActiveCfg = Debug|x86
+		{59250BAF-0000-0000-0000-000000000000}.Debug|Mixed Platforms.ActiveCfg = Debug|x86
+		{59250BAF-0000-0000-0000-000000000000}.Debug|Mixed Platforms.Build.0 = Debug|x86
+		{59250BAF-0000-0000-0000-000000000000}.Debug|x86.ActiveCfg = Debug|x86
+		{59250BAF-0000-0000-0000-000000000000}.Debug|x86.Build.0 = Debug|x86
+		{59250BAF-0000-0000-0000-000000000000}.Release|Any CPU.ActiveCfg = Release|x86
+		{59250BAF-0000-0000-0000-000000000000}.Release|Mixed Platforms.ActiveCfg = Release|x86
+		{59250BAF-0000-0000-0000-000000000000}.Release|Mixed Platforms.Build.0 = Release|x86
+		{59250BAF-0000-0000-0000-000000000000}.Release|x86.ActiveCfg = Release|x86
+		{59250BAF-0000-0000-0000-000000000000}.Release|x86.Build.0 = Release|x86
```
Comment (gbasood) [inline SpaceStation14.sln]: I have no idea why this was needed
And someone who knows more about configuring this might want to give their input

Comment (PJB3005) [reply SpaceStation14.sln]: I can only assume it's VS being dumb.

Comment (gbasood) [reply SpaceStation14.sln]: Actually, in the full history, it looks much more sane, so I'm not too worried about it

**Comment by PJB3005** [comment]
Well whatever, take DO NOT MERGE off the PR when you feel it's ready for review.

**Comment by gbasood** [comment]
It's as ready as it will be, but it's broken because SFML.
Might need to wait till after robust engine

**Comment by gbasood** [comment]
No longer crashes
Seems to be ready, not sure what else I could possibly test

**Comment by PJB3005** [comment]
Time to trigger sonar.

**Pull request merged**